### PR TITLE
Add image formats to `Faker\Provider\Image`

### PR DESCRIPTION
**Pull request by CalebDW**
### What is the reason for this PR?
The [placeholder.com](https://placeholder.com/) API allows for a few image formats, whereas `image()` and `imageUrl()` in `Faker\Provider\Image` defaulted to `png`.

- [x] Slight feature extension

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- [x] New features and changes are [documented - PR #66](https://github.com/FakerPHP/fakerphp.github.io/pull/66)

### Summary of changes
Added a static array to `Faker\Provider\Image` which contains allowable image formats (per Placeholder). Appended `$format` argument to both `Image::image()` and `Image::imageUrl()` which defaults to 'png'. Throw an exception in `Image::imageUrl()` if an invalid format is received. Wrote unit tests for all the image formats and `IllegalArgumentException`.

### Review checklist

- [x] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

File: src/Faker/Provider/Image.php [hunk 1]
```diff
@@ -22,6 +22,13 @@ class Image extends Base
         'fashion', 'people', 'nature', 'sports', 'technics', 'transport',
     ];
 
+    /**
+     * @var array
+     */
+    public static $imageFormats = [
```
Comment (pimjansen) [inline src/Faker/Provider/Image.php:28]: Can we add constants for this?

File: src/Faker/Provider/Image.php [hunk 2]
```diff
@@ -44,9 +52,19 @@ public static function imageUrl(
...
+        $gray = false,
+        $format = 'png'
     ) {
-        $size = sprintf('%dx%d.png', $width, $height);
+        // Validate image format
+        if (!in_array(strtolower($format), static::$imageFormats)) {
+            throw new \InvalidArgumentException(sprintf(
+                'Invalid image format "%s". Allowable formats are: %s',
+                $format,
+                implode(', ', static::$imageFormats)
```
Comment (pimjansen) [inline src/Faker/Provider/Image.php:63]: Implement the constants as reference here

File: src/Faker/Provider/Image.php [hunk 3]
```diff
@@ -101,10 +120,10 @@ public static function image(
         // Generate a random filename. Use the server address so that a file
         // generated at the same time on a different server won't have a collision.
         $name = md5(uniqid(empty($_SERVER['SERVER_ADDR']) ? '' : $_SERVER['SERVER_ADDR'], true));
-        $filename = $name . '.png';
+        $filename = $name . '.' . $format;
```
Comment (pimjansen) [inline src/Faker/Provider/Image.php:123]: Pls use sprintf instead of string concats

**Review by pimjansen — CHANGES_REQUESTED** [review]

File: src/Faker/Provider/Image.php [hunk 4]
```diff
@@ -90,7 +108,8 @@ public static function image(
         $fullPath = true,
         $randomize = true,
         $word = null,
-        $gray = false
+        $gray = false,
+        $format = 'png'
```
Comment (bram-pkg) [inline src/Faker/Provider/Image.php:112]:
```suggestion
        $format = 'png',
```

Comment (pimjansen) [reply src/Faker/Provider/Image.php:112]: This is only php 8+ @bram-pkg

Comment (bram-pkg) [reply src/Faker/Provider/Image.php:112]: Oh right, ugh.

Comment (CalebDW) [reply src/Faker/Provider/Image.php:112]: Is PHP 8+ not supported? I used named arguments in the unit tests.

Comment (pimjansen) [reply src/Faker/Provider/Image.php:112]: > Is PHP 8+ not supported? I used named arguments in the unit tests.

No it is 7.1+ see pipelines

File: src/Faker/Provider/Image.php (hunk 1 above)
Comment (CalebDW) [reply src/Faker/Provider/Image.php]: Added constants, apparently the `placeholder` website wouldn't actually generate a `gif` format, so I removed it.

File: src/Faker/Provider/Image.php (hunk 2 above)
Comment (CalebDW) [reply src/Faker/Provider/Image.php]: I implemented the constants in getter functions so I could access them from the unit tests. Let me know if this is acceptable.

File: src/Faker/Provider/Image.php (hunk 3 above)
Comment (CalebDW) [reply src/Faker/Provider/Image.php]: Done, may I ask why though?

**Comment by pimjansen** [comment]
Phpcs is failing. Can you review?

File: src/Faker/Provider/Image.php [hunk 5]
```diff
@@ -44,9 +49,20 @@ public static function imageUrl(
...
     ) {
-        $size = sprintf('%dx%d.png', $width, $height);
+        // Validate image format
+        $imageFormats = static::getFormats();
+        if (!in_array(strtolower($format), $imageFormats, true)) {
```
Comment (bram-pkg) [inline src/Faker/Provider/Image.php:57]:
```suggestion

        if (!in_array(strtolower($format), $imageFormats, true)) {
```

File: src/Faker/Provider/Image.php [hunk 6]
```diff
@@ -136,4 +153,18 @@ public static function image(
...
+    {
+        return array_keys(static::getFormatConstants());
+    }
+
+    public static function getFormatConstants(): array
+    {
+        return [
+            static::FORMAT_JPG => constant('IMAGETYPE_JPEG'),
+            static::FORMAT_JPEG => constant('IMAGETYPE_JPEG'),
+            static::FORMAT_PNG => constant('IMAGETYPE_PNG')
```
Comment (bram-pkg) [inline src/Faker/Provider/Image.php:167]:
```suggestion
            static::FORMAT_PNG => constant('IMAGETYPE_PNG'),
```

**Comment by CalebDW** [comment]
I fixed the above issues, but I am unable to run `phpcs` myself since I have PHP 8.1 installed on my system.

**Comment by pimjansen** [comment]
> I fixed the above issues, but I am unable to run `phpcs` myself since I have PHP 8.1 installed on my system. 

Pipeline is green, will do final review tomorrow so we can merge

**Review by pimjansen — APPROVED** [review]

**Comment by pimjansen** [comment]
LGTM

**Comment by jderusse** [comment]
Adding a method in an interface is a BC break. 
This PR should have been released in a Major version.

**Comment by DvDty** [comment]
> Adding a method in an interface is a BC break. This PR should have been released in a Major version.

Interesting, there is a job in the pipeline that checks for backwards compatibility and it has passed for the current pull request: https://github.com/FakerPHP/Faker/actions/runs/2293913195

**Pull request merged**